Sources/PackageModel/Target.swift

```diff
@@ -427,7 +428,8 @@ public final class ClangTarget: Target {
         dependencies: [Target.Dependency] = [],
         buildSettings: BuildSettings.AssignmentTable = .init()
     ) {
-        assert(includeDir.contains(sources.root), "\(includeDir) should be contained in the source root \(sources.root)")
+        
+        assert(includeDir.isDirectory && includeDir.contains(sources.root), "\(includeDir) should be contained in the source root \(sources.root)")
```

Comment:
Should this be an assertion or is this a user error?  Assertions are for programming errors, and are compiled out in release builds.

Reply:
Moved a check out of this and throw an error per below